terraform_compliance/steps/steps.py

```diff
@@ -364,7 +364,7 @@ def fail(condition, name=None):
 
     elif type(values) is dict:
         if 'values' in values:
-            if values['values'] is None and regex == '\x00' and condition == 'must not':
+            if values['values'] is None and regex == 'null' and condition == 'must not':
```

Comment:
`'null'` (string) is different than `null`/`\x00`. You can't replace `\x00` conditional with `'null'`, this test would fail with `null` values (not `'null'`).

This problem should not be solved like this. Instead there should be addition to an existing step which is checking this already (`its_value_condition_be null` step definition) on https://github.com/eerkunt/terraform-compliance/blob/fefd77b3fdee3d94f9b01a9e99b64f1ef4b83249/terraform_compliance/steps/steps.py#L390

e.g.

```python
its_value_condition_match_the_search_regex_regex(_step_obj, condition, u'^null$')
```

Reply:
you are right, there is already a step available to take care of the null conditions.
I guess I overlooked at the steps.py :( 
no changes required in steps.py

Reply:
We would also want to filter values that has string `null` in it. Thus, it is better to add

``python
its_value_condition_match_the_search_regex_regex(_step_obj, condition, u'^null$')
```
On the last step we have in `steps.py`

https://github.com/eerkunt/terraform-compliance/blob/fefd77b3fdee3d94f9b01a9e99b64f1ef4b83249/terraform_compliance/steps/steps.py#L390